--- a/libraries/Cabal/Cabal/Distribution/Simple/Configure.hs.orig	2018-03-05 00:45:45.000000000 +0900
+++ b/libraries/Cabal/Cabal/Distribution/Simple/Configure.hs	2018-10-25 15:14:51.786678408 +0900
@@ -1726,12 +1726,12 @@
                      -- should NOT be glomming everything together.)
                      ++ [ "-I" ++ buildDir lbi </> "autogen" ]
                      -- `configure' may generate headers in the build directory
-                     ++ [ "-I" ++ buildDir lbi </> dir | dir <- collectField PD.includeDirs
+                     ++ [ "-I" ++ buildDir lbi </> dir | dir <- ordNub (collectField PD.includeDirs)
                                                        , not (isAbsolute dir)]
                      -- we might also reference headers from the packages directory.
-                     ++ [ "-I" ++ baseDir lbi </> dir | dir <- collectField PD.includeDirs
+                     ++ [ "-I" ++ baseDir lbi </> dir | dir <- ordNub (collectField PD.includeDirs)
                                                       , not (isAbsolute dir)]
-                     ++ [ "-I" ++ dir | dir <- collectField PD.includeDirs
+                     ++ [ "-I" ++ dir | dir <- ordNub (collectField PD.includeDirs)
                                       , isAbsolute dir]
                      ++ ["-I" ++ baseDir lbi]
                      ++ collectField PD.cppOptions
@@ -1753,7 +1753,7 @@
                         | dep <- deps
                         , opt <- Installed.ccOptions dep ]
 
-        commonLdArgs  = [ "-L" ++ dir | dir <- collectField PD.extraLibDirs ]
+        commonLdArgs  = [ "-L" ++ dir | dir <- ordNub (collectField PD.extraLibDirs) ]
                      ++ collectField PD.ldOptions
                      ++ [ "-L" ++ dir
                         | dir <- ordNub [ dir
